### DISCUSSION TITLE
Is there any documentation which helps to change the target version to 33 in mobile app?

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#608] [Documentation] [answered]
Hello all,

I'm working on SunbirdEd mobile app (4.6 version).
My current `defaultTargetSdkVersion=31
      defaultCompileSdkVersion=31 ` is 31.

As per the **Google Play requires all apps to meet target API level requirements before Aug 31, 2023** I need to change the version to 33.

Is there any documentation available or can anyone help me how to achieve that?

@swayangjit @rajeevsathish

**Comment by swayangjit** [comment]
@nidhikachhawa24 We are working on the documentation and will update you soon.

**Reply by nidhikachhawa24** [reply]
@swayangjit Thank you

**Comment by swayangjit** [comment, marked as answer]
@nidhikachhawa24 Earlier forgot to inform you that we are providing a patch in release-5.1.0. So you will have to take the code manually and apply the patch if you are using release-4.6.0. The documentation is available [here](https://project-sunbird.atlassian.net/wiki/spaces/SUN/pages/3354329090/Migration+of+Sunbird+app+from+to+android+SDK+level+31+to+SDK+level+33).

**Reply by siddhishinde0723** [reply]
@swayangjit I have followed the instructions in the document, but there is still a problem.

Issue : Wrote out android application name "@string/_app_name" to /home/ttpl-rt-140/Project/SunbirdSaas/SunbirdEd-mobile-app/platforms/android/app/src/main/res/values/strings.xml
Cannot read properties of null (reading 'find')
TypeError: Cannot read properties of null (reading 'find')

 I'm using the versions listed below.

Ionic:

   Ionic CLI                     : 7.1.1 (/home/ttpl-rt-140/.nvm/versions/node/v16.20.2/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.7.0
   @angular-devkit/build-angular : 0.901.15
   @angular-devkit/schematics    : 9.1.15
   @angular/cli                  : 9.1.15
   @ionic/angular-toolkit        : 2.3.3

Cordova:

   Cordova CLI       : 12.0.0 (cordova-lib@12.0.1)
   Cordova Platforms : android 12.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.2.1, (and 47 other plugins)

Utility:

   cordova-res : 0.15.4
   native-run  : 1.7.2

System:

   Android SDK Tools : 26.1.1 (/home/ttpl-rt-140/Android/Sdk)
   NodeJS            : v16.20.2 (/home/ttpl-rt-140/.nvm/versions/node/v16.20.2/bin/node)
   npm               : 8.19.4
   OS                : Linux 5.15

**Reply by swayangjit** [reply]
@siddhishinde0723 Could you let me know if you removed and then added your platform?

**Reply by siddhishinde0723** [reply]
@swayangjit Yes have removed my platform and then i done the steps

**Reply by swayangjit** [reply]
@siddhishinde0723  can you do ececute build.sh for the same ?

**Reply by siddhishinde0723** [reply]
@swayangjit Yes i did still getting the same issue

**Reply by swayangjit** [reply]
@siddhishinde0723 Let me know your available timing we will connect and sort it out.

**Reply by siddhishinde0723** [reply]
@swayangjit is 12pm  fine for you

**Reply by siddhishinde0723** [reply]
Check out the document to switch from 31 to 33 for release 4.6.0. [view](https://docs.google.com/document/d/e/2PACX-1vT1p76PJpaaq4ez0CgH6AlhjBZHradwZXFCBzZ3k6Y86S8CuA32MfnHQoTu_1mi3Ti7hMaOyMJiLnuH/pub
)

**Reply by swayangjit** [reply]
@siddhishinde0723  Please do provide a public link

**Comment by siddhishinde0723** [comment]
@swayangjit As we have migrated from 31 to 33 the apk works fine with android version 12 and above but the apk wont work below 12. We are facing issue for installing apk on android version 11, 10 any solution?
I have tried changing minsdkversion to 31,30 still it wont work.

**Reply by swayangjit** [reply]
@siddhishinde0723  Can you let us know what is the error that you are getting?

**Reply by siddhishinde0723** [reply]
@swayangjit We are getting this error "there was a problem parsing the package"

**Reply by swayangjit** [reply]
@siddhishinde0723 Install it via ADB and check what the error is

**Reply by siddhishinde0723** [reply]
@swayangjit  This is the error we get on ADB i am using Android version 11 
"Error code: 'INSTALL_FAILED_OLDER_SDK', message='INSTALL_FAILED_OLDER_SDK: Failed parse during installPackageLI: /data/app/vmdl1979771308.tmp/base.apk (at Binary XML file line #8): Requires newer sdk version #31 (current version is #30)'

**Reply by swayangjit** [reply]
@siddhishinde0723  we have tried installing sunbird app It installing in android 10 and 11 devices. Any other changes you did apart from SDK migration ?

**Reply by siddhishinde0723** [reply]
@swayangjit  No i haven't done any changes rather than SDK

**Reply by swayangjit** [reply]
@siddhishinde0723 Can you please share the apk ?

**Comment by siddhishinde0723** [comment]
Problem: Even though we had adjusted the minSdk version to 22, the apk was still using 31. Installing the apk on an Android version lower than 12 was producing an error. 

Solution: Sunbird.properties files also need to have the minSdk version changed. It was reading the SDK version from the properties file while creating the APK.

**Reply by swayangjit** [reply]
@siddhishinde0723 Can you please put up a sample as in sunbird.properties there is no property with minSDK.
